### PR TITLE
Fix some log levels and reduce (debug) log spam

### PR DESCRIPTION
**Pull request by uklotzde**
While testing I noticed that we still write too many debug logs that should only be written while debug logging is enabled explicitly.

I've divided the modifications into commits with appropriate commit messages for easy verification.

### CODE REVIEW

File: src/sources/metadatasourcetaglib.cpp [hunk 1]
```diff
@@ -745,9 +749,11 @@ class SafelyWritableFile final {
 std::pair<MetadataSource::ExportResult, QDateTime>
 MetadataSourceTagLib::exportTrackMetadata(
         const TrackMetadata& trackMetadata) const {
-    kLogger.debug() << "Exporting track metadata"
-            << "into file" << m_fileName
-            << "with type" << m_fileType;
+    if (kLogger.debugEnabled()) {
+        kLogger.debug() << "Exporting track metadata"
```
Comment (daschuer) [inline src/sources/metadatasourcetaglib.cpp]: I am not sure if the file name is logged somewhere else as well. If not this one is important to identify a faulty file that might be the cause for a taglib crash.

File: src/sources/soundsourceproxy.cpp [hunk 2]
```diff
@@ -618,12 +620,14 @@ mixxx::AudioSourcePointer SoundSourceProxy::openAudioSource(const mixxx::AudioSo
...
     while (m_pSoundSource && !m_pAudioSource) {
-        kLogger.debug() << "Opening file"
-                << getUrl().toString()
-                << "with provider"
-                << getSoundSourceProvider()->getName()
-                << "using mode"
-                << openMode;
+        if (kLogger.debugEnabled()) {
+            kLogger.debug() << "Opening file"
+                    << getUrl().toString()
```
Comment (daschuer) [inline src/sources/soundsourceproxy.cpp]: This one can be also a helper for identifying a crashing file.

**Comment by daschuer** [comment]
I think we should keep at least the file related messages to identify a faulty file.

**Comment by uklotzde** [comment]
Ok. I've re-enabled the two file related logs and also added comments why we decided do this.

**Comment by daschuer** [comment]
Thank you. Unfortunately Mac build fails 

```
Undefined symbols for architecture x86_64:
  "mixxx::Logging::s_logLevel", referenced from:
      mixxx::Logging::enabled(mixxx::LogLevel) in metadatasourcetaglib.os
      mixxx::Logging::enabled(mixxx::LogLevel) in trackmetadatataglib.os
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
scons: *** [osx64_build/plugins/osx64_build/m4a/libsoundsourcem4a.dylib] Error 1
scons: building terminated because of errors.
```

**Comment by daschuer** [comment]
Do you have a idea to fix the builds soon, or should we remove this from the 2.1.1 milestone?

**Comment by daschuer** [comment]
LGTM, Is this still [WIP]?

**Comment by uklotzde** [comment]
No, I just wanted to await the build results. Building locally with clang on Linux worked, I was not able to reproduce the linker error.

Unfortunately I had to bump the API version. This is a really ugly hack and only a partial fix. But we can't fix the whole logging with its global state variables that don't get initialized properly in the plugins. We really need to get rid of the plugins.

**Comment by daschuer** [comment]
OK, the CI fails unrelated. Merge?

**Comment by uklotzde** [comment]
There is still a small chance that the Windows build might fail. Although according to my understanding this new version should at least be _less broken_ than before ;)

**Comment by uklotzde** [comment]
I need to test it again. Currently the plugins fail to load even on Linux!?!

File: src/sources/soundsourcepluginapi.h [hunk 3]
```diff
@@ -3,6 +3,7 @@
 
 #define MIXXX_SOUNDSOURCEPLUGINAPI_VERSION 17
```
Comment (ninomp) [inline src/sources/soundsourcepluginapi.h]: Shouldn't this be 18?

**Comment by uklotzde** [comment]
@ninomp Thanks for the hint. Not the first time that this happened.

**Comment by daschuer** [comment]
Ok then let's merge it, and see what happens.

**Comment by daschuer** [comment]
This was one of the last 2.1.1 PRs.
Would you mind to look into: https://github.com/mixxxdj/mixxx/pull/1691?

**Pull request merged**